src/auth/get-permissions.js

```diff
@@ -1,98 +1,25 @@
-  // Cannot be retrieved in a single call so need to make multiple calls to different APIs
-  // These calls are authenticated using the token returned from Defra Identity
-  // All APIs are accessible via a series of RESTful endpoints hosted in Crown Hosting
-  // For the purposes of this example, we will simulate these calls using mock data
-  // 1. Get personId from RPS API
-  const personId = await getPersonId({ crn, token })
-  // 2. Get roles and privileges from Siti Agri API
-  const { role, privileges } = await getRolesAndPrivileges(personId, organisationId, { crn, token })
-  // 3. Map roles and privileges to scope
-  // An application specific permission is added to demonstrate how to add local, non-Siti Agri permissions
-  const scope = [DEFAULT_SCOPE, ...privileges]
-  // Hapi.js assumes permissions are added in a `scope` array
-  return { role, scope }
+import { dalConnector } from '../dal/connector.js'
+import { permissionsQuery } from '../dal/queries/permissions-query.js'
+import { mapPermissions } from '../mappers/permissions-mapper.js'
+import { config } from '../../src/config/index.js'
+import { mappedData } from '../mock-data/mock-permissions.js'
+
+async function getPermissions (sbi, crn, email) {
+  const permission = config.get('featureToggle.dalConnection') ? await getFromDal(sbi, crn, email) : mappedData
+  return permission
 }
 
-async function getPersonId (headers) {
-  // simulate call to RPS API via common DAL
-  // Only id is needed for mapping roles, but other fields shown for context for what else is available
-  // PATH: /person/3337243/summary
-  // METHOD: GET
-  // HEADERS:
-  //   crn: <headers.crn>
-  //   Authorization <headers.token>
-
-  const mockResponse = {
-    headers,
-    data: {
-      id: '123456',
-      customerReferenceNumber: '1234567890', // crn
-      title: 'Mr',
-      firstName: 'Andrew',
-      lastName: 'Farmer',
-      landline: '01234567890',
-      mobile: '01234567890',
-      email: 'a.farmer@farms.com',
-      address: {
-        address1: 'Address line 1',
-        address2: 'Address line 2',
-        address3: 'Address line 3',
-        address4: 'Address line 4',
-        address5: 'Address line 5',
-        city: 'City',
-        county: 'County',
-        postcode: 'FA1 1RM',
-        country: 'UK'
-      },
-      doNotContact: false,
-      locked: false
-    }
-  }
+const getFromDal = async (sbi, crn, email) => {
+  const variables = { sbi, crn }
 
-  return mockResponse.data.id
-}
+  const dalResponse = await dalConnector(permissionsQuery, variables, email)
 
-async function getRolesAndPrivileges (personId, organisationId, { headers }) {
-  // simulate call to Siti Agri API
-  // returns all roles and privileges for so need to filter for logged in user
-  // PATH: /SitiAgriApi/authorisation/organisation/<organisationId>/authorisation
-  // METHOD: GET
-  // HEADERS:
-  //   crn: <headers.crn>
-  //   Authorization <headers.token>
+  if (dalResponse.data) {
```

Comment:
If the DAL request fails within `dalConnector` then the error is caught and the value of `dalResponse` will be undefined.

That means the error will manifest here when we try to do `undefined.data` which will result in an error message unrelated to the actual issue.

Suggest either handling the error appropriately within the `dalConnector` or checking if the response is undefined here.  The former naturally feels right to me as that's where the error occurred, but there maybe wider application context I'm not aware of that may influence that decision.